dove/src/transaction.rs

```diff
@@ -0,0 +1,875 @@
+        return Err(anyhow!("Invalid call script format.\
+             Expected function identifier. Use pattern \
+             'script_name<comma separated type parameters>(comma separated parameters WITHOUT signers)'"));
+    }
+
+    let script_name = lexer.content().to_owned();
+
+    lexer.advance().map_err(map_err)?;
+
+    let type_parameters = if lexer.peek() == Tok::Less {
+        let mut type_parameter = vec![];
+
+        lexer.advance().map_err(map_err)?;
+        while lexer.peek() != Tok::Greater {
+            if lexer.peek() == Tok::EOF {
+                return Err(anyhow!("Invalid call script format.\
+             Invalid type parameters format.. Use pattern \
+             'script_name<comma separated type parameters>(comma separated parameters WITHOUT signers)'"));
+            }
+
+            if lexer.peek() == Tok::Comma {
+                lexer.advance().map_err(map_err)?;
+                continue;
+            }
+
+            type_parameter.push(parse_type_params(&mut lexer)?);
+        }
+        lexer.advance().map_err(map_err)?;
+        type_parameter
+    } else {
+        vec![]
+    };
+
+    if lexer.peek() != Tok::LParen {
+        return Err(anyhow!("Invalid call script format.\
+             Invalid script arguments format.. Left paren '(' is expected. Use pattern \
+             'script_name<comma separated type parameters>(comma separated parameters WITHOUT signers)'"));
+    }
+
+    let mut arguments = vec![];
+
+    lexer.advance().map_err(map_err)?;
+    while lexer.peek() != Tok::RParen {
+        if lexer.peek() == Tok::EOF {
+            return Err(anyhow!("Invalid call script format.\
+             Invalid arguments format.. Use pattern \
+             'script_name<comma separated type parameters>(comma separated parameters WITHOUT signers)'"));
+        }
+
+        if lexer.peek() == Tok::Comma {
+            lexer.advance().map_err(map_err)?;
+            continue;
+        }
+
+        let mut token = String::new();
+        token.push_str(lexer.content());
+        let sw = lexer.peek() == Tok::LBracket;
+        lexer.advance().map_err(map_err)?;
+        if sw {
+            while lexer.peek() != Tok::RBracket {
+                token.push_str(lexer.content());
+                lexer.advance().map_err(map_err)?;
+            }
+            token.push_str(lexer.content());
+        } else {
+            while lexer.peek() != Tok::Comma && lexer.peek() != Tok::RParen {
+                token.push_str(lexer.content());
+                lexer.advance().map_err(map_err)?;
+            }
+        }
+        arguments.push(token);
+        if !sw && lexer.peek() == Tok::RParen {
+            break;
+        }
+        lexer.advance().map_err(map_err)?;
+    }
+
+    Ok((script_name, type_parameters, arguments))
+}
+
+/// parse type parаms
+/// <u8,u64> => vec![TypeTag::U8,TypeTag::U64]
+pub fn parse_type_params(lexer: &mut Lexer) -> Result<TypeTag, Error> {
```

Comment:
I don't believe function name and doc-comment are reflecting this particular function, given significant difference in signature (`parse_param`_**s**_ can't return a single `TypeTag`).